retry/tripperware.go

```diff
@@ -26,7 +26,8 @@ func Tripperware(opts ...Option) httpwares.Tripperware {
 			if !o.decider(req) && !isEnabled(req.Context()) {
 				return next.RoundTrip(req)
 			}
-			if o.maxRetry == 0 || getBody(req) == nil {
+			getBodyFn := getBody(req)
+			if o.maxRetry == 0 || getBodyFn == nil {
```

Comment:
The go1.7 `getBody` returns nil if the original body is nil, meaning body-less requests won't be retried with that code. I think you'll want to fix that too? See if there's a test.

Reply:
Good spot :) Fortunately the tests also caught this